src/features/MainScreen/index.tsx

```diff
@@ -30,6 +30,29 @@ export const MainScreen: FC = () => {
 		});
 	}, [tagsRegistry, updateNotes]);
 
+	useEffect(() => {
+		const preventTextInsert = (e: MouseEvent) => {
+			if (e.button !== 1) return;
+
+			const isEditable = e.composedPath().some((element) => {
+				return (
+					element instanceof HTMLInputElement ||
+					element instanceof HTMLTextAreaElement ||
+					(element instanceof HTMLElement && element.isContentEditable)
```

Comment:
This code potentially may run frequently, so 3 times to check for proto per run sounds expensive.
We could replace it to check a `tagName` for `INPUT` and `TEXTAREA`.

Check for element may be replaced via check `nodeType` for `Node.ELEMENT_NODE` https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType